### PR TITLE
[TEST] Fix flaky test in TPCHQuerySuite

### PR DESCRIPTION
**Pull request by cxzl25**
# :mag: Description
## Issue References 🔗

This pull request fixes #

## Describe Your Solution 🔧

TPCHQuerySuite:
- run query on tiny *** FAILED ***
  q14 (TPCHQuerySuite.scala:74)

https://github.com/apache/kyuubi/actions/runs/6891462902/job/18750551743?pr=5716

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklists
## 📝 Author Self Checklist

- [ ] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [ ] I have performed a self-review
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [x] Pull request title is okay.
- [x] No license issues.
- [ ] Milestone correctly set?
- [ ] Test coverage is ok
- [ ] Assignees are selected.
- [ ] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: extensions/spark/kyuubi-spark-connector-tpch/src/main/resources/kyuubi/tpch/q14.sql [hunk 1]
```diff
@@ -17,11 +17,11 @@
 -- using default substitutions
 
 select
-    100.00 * sum(case
+    round(100.00 * sum(case
         when p_type like 'PROMO%'
             then l_extendedprice * (1 - l_discount)
         else 0
-    end) / sum(l_extendedprice * (1 - l_discount)) as promo_revenue
+    end) / sum(l_extendedprice * (1 - l_discount)), 2) as promo_revenue
```
Comment (cxzl25) [inline extensions/spark/kyuubi-spark-connector-tpch/src/main/resources/kyuubi/tpch/q14.sql:24]: GA
```
query q14 result: List([15.486545812284065])
```

Local Mac
```
query q14 result: List([15.486545812284069])
```

File: extensions/spark/kyuubi-spark-connector-tpch/src/test/scala/org/apache/kyuubi/spark/connector/tpch/TPCHQuerySuite.scala [hunk 2]
```diff
@@ -59,20 +59,15 @@ class TPCHQuerySuite extends KyuubiFunSuite {
...
-          val schemaDDL = LICENSE_HEADER + schema.toDDL + "\n"
-          spark.createDataFrame(result.toList.asJava, schema).createTempView(s"$name$viewSuffix")
-          val sumHashResult = LICENSE_HEADER + spark.sql(
-            s"select sum(hash(*)) from $name$viewSuffix").collect().head.get(0) + "\n"
-          val tuple = generateGoldenFiles("kyuubi/tpch", name, schemaDDL, sumHashResult)
-          assert(schemaDDL == tuple._1)
-          assert(sumHashResult == tuple._2)
-        } catch {
-          case cause: Throwable =>
-            fail(name, cause)
```
Comment (cxzl25) [inline extensions/spark/kyuubi-spark-connector-tpch/src/test/scala/org/apache/kyuubi/spark/connector/tpch/TPCHQuerySuite.scala:74]: Cannot display the comparison results of assert

File: extensions/spark/kyuubi-spark-connector-tpch/src/test/scala/org/apache/kyuubi/spark/connector/tpch/TPCHQuerySuite.scala [hunk 3]
```diff
@@ -31,12 +31,12 @@ import org.apache.kyuubi.spark.connector.common.LocalSparkSession.withSparkSessi
 /**
  * To run this test suite:
  * {{{
- *   KYUUBI_UPDATE=0 dev/gen/gen_tpcdh_queries.sh
+ *   KYUUBI_UPDATE=0 dev/gen/gen_tpch_queries.sh
  * }}}
  *
  * To re-generate golden files for this suite:
  * {{{
- *   dev/gen/gen_tpcdh_queries.sh
```
Comment (cxzl25) [inline extensions/spark/kyuubi-spark-connector-tpch/src/test/scala/org/apache/kyuubi/spark/connector/tpch/TPCHQuerySuite.scala:39]: typo

**Review by pan3793 — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]

**Comment by cxzl25** [comment]
Thanks, merged to master/1.8.

**Pull request closed without merging**